python-threatexchange/threatexchange/signal_type/tests/test_hash_from_x.py

```diff
@@ -4,29 +4,54 @@
 import tempfile
 from PIL import Image
 
-
-from threatexchange.content_type import meta
-from threatexchange.signal_type.signal_base import FileHasher, StrHasher
-from pathlib import Path
+from threatexchange.signal_type import (
+    md5,
+    pdq_ocr,
+    pdq,
+    raw_text,
+    tlsh,
+    trend_query,
+    url_md5,
+    url,
+)
+from threatexchange.signal_type.signal_base import FileHasher, TextHasher
 
 
 class SignalTypeHashTest(unittest.TestCase):
     """
-    If a SignalType is a FileHasher/StrHasher it should be able to
-    take the corresponding content medium without throwing.
+    Sanity check for signal type hashing methods.
     """
 
-    def test_file_hashers_have_impl(self):
-        signal_types = [s() for s in meta.get_all_signal_types()]
-        file_hashers = [s for s in signal_types if isinstance(s, FileHasher)]
-        fp = tempfile.NamedTemporaryFile(suffix=".jpg")
-        image = Image.new("RGB", size=(50, 50), color=(155, 0, 0))
-        image.save(fp, "jpeg")
-        for file_hasher in file_hashers:
-            file_hasher.hash_from_file(fp.name)
+    # TODO - maybe make a metaclass for this to automatically detect?
+    SIGNAL_TYPES_TO_TEST = [
+        md5.PhotoMD5Signal,
+        md5.VideoMD5Signal,
+        pdq_ocr.PdqOcrSignal,
+        pdq.PdqSignal,
+        raw_text.RawTextSignal,
+        tlsh.TLSHSignal,
+        trend_query.TrendQuerySignal,
+        url_md5.UrlMD5Signal,
+        url.URLSignal,
+    ]
+
+    def test_signal_names_unique(self):
+        seen = {}
+        for s in self.SIGNAL_TYPES_TO_TEST:
+            name = s.get_name()
+            assert (
+                name not in seen
+            ), f"Two signal types share the same name: {s!r} and {seen[name]!r}"
+
+    def test_signal_types_have_content(self):
+        for s in self.SIGNAL_TYPES_TO_TEST:
+            assert s.get_content_types(), "{s!r} has no content types"
 
     def test_str_hashers_have_impl(self):
-        signal_types = [s() for s in meta.get_all_signal_types()]
-        str_hashers = [s for s in signal_types if isinstance(s, StrHasher)]
-        for str_hasher in str_hashers:
-            str_hasher.hash_from_str("test string")
+        text_hashers = [
```

Comment:
The dumb answer:
* It was already here and I just refactored it

The marginally smarter answer:
* I don't think checking that content types are non-empty is typeable, but could be wrong
* ABC is an option for checking that an implementation exists, but just checking that the implementation is callable can shake out a couple of spiders.
* I'm a fan of smoke testing as opposed to no testing, and since we're low, I'd prefer to keep it in for now